### PR TITLE
#166272290 Setup End to End Testing

### PR DESCRIPTION
**Pull request by GerrardE**
#### What does this PR do?
Set up end to end testing for the application

#### Description of Task to be completed?
- Install and configure cypress for end to end testing
- Write sample tests

#### How should this be manually tested?
1. clone this repo
2. run "npm install"
3. run "npm run cypress:all" or "npm run cypress:open" 

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166272290](https://www.pivotaltracker.com/story/show/166272290)

#### Screenshots (if appropriate)
Screenshot 2019-07-05 at 7 22 11 PM

#### Questions:
N/A

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+  cypress: cypress-io/cypress@1
 jobs: # a collection of steps
   build: # runs not using Workflows must have a `build` job as entry point
     environment:
       CC_TEST_REPORTER_ID: ${TEST_REPORTER_ID}
+      TERM: xterm
     working_directory: ~/authors-haven # directory where steps will run
     docker:
-      - image: circleci/node:10.3.0
+      - image: cypress/base:8
```
Comment (fidelisojeah) [inline .circleci/config.yml]: lol what are you even doing?

please create a new job and move your tests there

**Review by despeauxz — DISMISSED** [review]

**Review by EMacco — DISMISSED** [review]
Great job. Tests work fine.

**Review by despeauxz — DISMISSED** [review]

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (GerrardE) [reply .circleci/config.yml]: I have implemented this feedback, Thanks

**Review by despeauxz — APPROVED** [review]

**Review by adex001 — APPROVED** [review]
Looks good and works well.

**Review by fidelisojeah — APPROVED** [review]

**Pull request merged**